pweave/scripts.py

```diff
@@ -1,6 +1,15 @@
 import sys
 from optparse import OptionParser
 import pweave
+import logging
+
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+logging.getLogger("ipykernel").propagate = False
```

Comment:
line should be split into two (with `ipykernel_logger` intermediate variable)

Reply:
It's three lines and the change you requested is not clear to me.

Reply:
I am wary of any change that increases program length and create names that are used only once. But if you define ipykernel_logger here and then reuse at line 66 & 68 then I support this change. If you redefine ipykernel_logger three times, then I don't see the advantage.

Reply:
@piccolbo I try to avoid what is called "trainwreck pattern" if I remember correctly. Get to go, may elaborate a bit more tomorrow.

Reply:
I googled it myself. Other people call it "fluent style" and build whole packages around it (e.g. https://github.com/olirice/flupy). The "train wreck pattern" is an argument against composition and as such we should reject it. http://wiki.c2.com/?TrainWreck Because something has the word pattern attached to it, it doesn't mean we should believe it. What is the difference between f(g(x)) and x.g().f()? And how is z = g(x); f(z) superior? But if you use it 10 times, or even three times, then sure, maybe there is an abstraction possibility there. But that's true of every repeating snippet of code, not just chained methods. That's red herring.

Reply:
As it is being used 3 times, it probably should go to a separate function with meaningful name. ;)

Nevertheless, I did not requested the change because it is a style with an unpleasant name.
I requested it because it was not obvious to me what that line of code does.
I believe that readibility counts - not because it is written in PEP 20, but because for me code is being written mostly for humans (to convey intent).
Even if computers do not understand the intent (we have a bug), a human can fix the bug easily if the code is clean.

The main problem with "fluent style" is that (for me) it is hard to read. The `f(g(x, y))` for the interpreter may be same as `x.g(y).f()` if they are formally the same. But for me the former is a mathematical notation I am familiar with while the latter forces me to think about the return type of g. It may be not a problem for someone familiar with "fluent notation" though.
The real problem starts when `.f()` alters object returned by `.g()`, as in this case of assignment.
For me it is counterintuitive. When the left hand side of assignment is a function/method call, it is too brainy for me. ;)
I like to have the "getters" and "setters" separated at least by a new line character.

As a side note - "fluent style" is also considered a "code smell" indicating that probably the "Law of Demeter" is being violated. Which makes me anxious. ;)

Reply:
The lines are perfectly readable to me. I was a human the last time I checked. Fluent style is just composition of methods, you should get yourself familiar with it. Many people prefer it because it follows the order of execution. For instance R and F# have pipeline notation so that f(g(x)) can be written as x|g|f without changing a line in f or g, and with the f(g(x)) notation still available. You are trying to force personal tastes on the project as if they were some fundamental rules of software engineering. Breaking chain of methods does not undo side effects. Side effects are the problem, not composition. I don't want to spend time arguing, I think I will focus again on my fork.

Reply:
@piccolbo It is possible that my mental capacity is not as big as yours, then. Nevertheless I am familiar with "train wreck/fluent style" - for some time I have been using it in JavaScript (jQuerry). But we are coding Python here - not jQuerry, R, F# etc.

Also I am not forcing my personal taste on anyone here (however I may advocate it). I am simply not taking responsibility for accepting code that I would not accept in my own project. For that I need an explicit @mpastell's consent.

In case of style my decision is indeed primarly opinion based. One reason for which I have not enough time for _pweave_ project is dealing with dirty code (including my own). Every single time the problem is not following advice of R.C.Martins, who discouraged "train wrecks" as a rule of thumb. With that _argumentum ab auctoritate_ (against your _argumentum ad populum_) I also consider the argument finished.